src/vm/arm/asmhelpers.S

```diff
@@ -441,7 +444,7 @@ LOCAL_LABEL(UMThunkStub_WrongAppDomain):
         ldr                 r2, [r3, #UMThunkMarshInfo__m_cbActualArgSize]
         cbz                 r2, LOCAL_LABEL(UM2MThunk_WrapperHelper_ArgumentsSetup)
 
-        add                 r0, r5, #UMThunkStub_StackArgs // Source pointer
+        add                 r0, r5, #UMThunkStub_StackArgsOffset // Source pointer
```

Comment:
Because, `UM2MThunk_WrapperHelper` is another function, it is NOT affected by the `r7` value of `UMThunkStub`. There is no reason for the argument of `UM2MThunk_WrapperHelper`, `r0`, be altered by the correction of `r7` in `UMThunkStub`.

Thus, the value for this line should be kept unchanged (`UMThunkStub_StackArgsSize`)